src/js/gi/containers/GiBillApp.jsx

```diff
@@ -16,18 +17,60 @@ const Disclaimer = () => {
 };
 
 class GiBillApp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.exitPreviewMode = this.exitPreviewMode.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.fetchConstants(this.props.location.query.version);
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      preview,
+      location: { query: { version: uuid } }
+    } = this.props;
+
+    const shouldHidePreviewMode = preview.display && !uuid;
+    const shouldShowPreviewMode = !preview.display && uuid && preview.version.createdAt;
+
+    if (shouldHidePreviewMode) {
+      this.props.exitPreviewMode();
+    } else if (shouldShowPreviewMode) {
```

Comment:
I did a double take here. I thought, "Wait, if it shouldn't show and it shouldn't hide, what should it do?" It wasn't until I went back to read the definitions of these variables that I figured out the purpose.

You might change these to `shouldExitPreviewMode` and `shouldEnterPreviewMode` for clarity. Then it's easy to figure out that if we shouldn't exit and we shouldn't enter, we're just staying where we're at.